firestore/integration_test_internal/src/firestore_test.cc

```diff
@@ -87,8 +96,22 @@ TEST_F(FirestoreIntegrationTest, GetInstance) {
   delete auth;
 }
 
+TEST_F(FirestoreTest, GetInstanceWithNamedDatabase) {
+  App* app = this->app();
+  EXPECT_NE(nullptr, app);
+
+  InitResult result;
+  Firestore* instance = Firestore::GetInstance(app, "foo", &result);
```

Comment:
Please use a `unique_ptr` here, thereby avoiding memory leak and `delete instance` can be removed since destructor is handled by `unique_ptr`.

Reply:
Used the `TestFirestoreWithDatabaseId` helper method to use `unique_ptr` behind the curtain.

Reply:
change to use `unique_ptr` instead of `TestFirestoreWithDatabaseId`